services/QuillLMS/app/models/activity.rb

```diff
@@ -153,6 +154,11 @@ class Activity < ApplicationRecord
 
   DEFAULT_MAX_GRADE_LEVEL = 12
 
+  def classification=(value)
+    super(value)
+    self.repeatable = false if value == ActivityClassification.diagnostic
+  end
```

Comment:
I didn't see a spec for this behavior, and it feels like it might be worth making one:
```
let(:diagnostic_classification) { create(:diagnostic) }
let(:repeatable) { true }
let(:activity) { create(:activity, repeatable:) }

it do
  expect { activity.classification = diagnostic_classification }
    .to change(activity.reload, :repeatable)
    .from(repeatable)
    .to(false)
end
```